### PR TITLE
[ContactStructuralMechanicsApplication] Clean up VariableComponentType

### PR DESCRIPTION
**Pull request by loumalouomega**
Basically removes the leftovers of VariableComponent (I think this is also the reason compilation fails, as this does not exist in the base class anymore and fails when linking)

### CODE REVIEW

**Review by philbucher — DISMISSED** [review]
Lets hope this is the cause of the error

Thx for cleaning

**Comment by loumalouomega** [comment]
> Lets hope this is the cause of the error
> 
> Thx for cleaning

Nope @roigcarlo any idea, I have not changed anything in the app in weeks, I think this is something we have modified in the core recently...

**Comment by loumalouomega** [comment]
In my machine it compiles bot regular and cotire

**Comment by philbucher** [comment]
Compilation was always a problem bcs of high memory consumption 
Can you exclude some files from cotire?

**Comment by philbucher** [comment]
The error says that the compiler is out of memory
Probably excluding the conditions could solve the problem

**Comment by philbucher** [comment]
Excluding the app from cotire completely works as expected

**Comment by loumalouomega** [comment]
> Excluding the app from cotire completely works as expected

It was a test

File: applications/ContactStructuralMechanicsApplication/CMakeLists.txt [hunk 1]
```diff
@@ -63,6 +63,7 @@ endif(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
 if(USE_COTIRE MATCHES ON)
     ## The interface_preprocess is not compatible with cotire (solve problem with explicit instantations)
     set_source_files_properties (${CMAKE_CURRENT_SOURCE_DIR}/custom_utilities/interface_preprocess.cpp PROPERTIES COTIRE_EXCLUDED TRUE)
+    set_source_files_properties (${CMAKE_CURRENT_SOURCE_DIR}/custom_conditions/*.cpp PROPERTIES COTIRE_EXCLUDED TRUE)
```
Comment (philbucher) [inline applications/ContactStructuralMechanicsApplication/CMakeLists.txt]: This doesn't work I think
=> guess you need to loop the files and set the properties explicitly

Comment (philbucher) [reply applications/ContactStructuralMechanicsApplication/CMakeLists.txt]: sth like 
~~~
file(GLOB condition_files ${CMAKE_CURRENT_SOURCE_DIR}/custom_conditions/*.cpp)

foreach(condition_file ${condition_files})
set_source_files_properties (condition_file PROPERTIES COTIRE_EXCLUDED TRUE)
endforeach(condition_file)
~~~

**Comment by loumalouomega** [comment]
@philbucher finally works

**Review by GuillermoCasas — APPROVED** [review]
It looks fine! And it is urgent, so I am approving

**Comment by philbucher** [comment]
cool!

I think it would be still a good thing to change it to the loop to automatically do this for all the conditions
Its also cleaner
can you do it please @loumalouomega ?

**Comment by philbucher** [comment]
> cool!
> 
> I think it would be still a good thing to change it to the loop to automatically do this for all the conditions
> Its also cleaner
> can you do it please @loumalouomega ?

ping @loumalouomega what do you think?

**Comment by loumalouomega** [comment]
> > cool!
> > I think it would be still a good thing to change it to the loop to automatically do this for all the conditions
> > Its also cleaner
> > can you do it please @loumalouomega ?
> 
> ping @loumalouomega what do you think?

It can be done

**Pull request merged**